### DISCUSSION TITLE
How to apply bold to regex match?

### DISCUSSION THREAD

**Discussion by nedvoid** [ianstormtaylor/slate#4778] [Q&A]
[stackoverflow post](https://stackoverflow.com/questions/70629704/slatejs-apply-bold-to-regex-match)

Trying to detect markdown and apply marks using [this answer](https://stackoverflow.com/a/67855927) from stackoverflow.

Currently I am getting the expected results but the following error occurs:
```
Could not completely normalize the editor after 126 iterations! This is usually due to incorrect normalization logic that leaves a node in an invalid state.
```

**How can I fix this?**

**Comment by TheSpyder** [comment]
Doing this in normalize seems like a bad idea, I’d do it on keydown personally, but that error means either the model is invalid (as the error says) or your normalize function is continually making changes. After the emoji has been inserted the text node becomes dirty again, so it’s essential normalize does nothing the second time.

I have two suggestions to track this down:
- log the entire model after the error and see if there are any bits you aren’t expecting. Also check that the model is valid (feel free to post it here if you aren’t sure).
- if the model is correct, then try logging inside the the function (unfortunately it will log at least 126 times) and/or put a debug break point in. That will be the easiest way to find out why it continually makes changes.

**Comment by TheSpyder** [comment]
Oh I missed the link to your stack overflow question. There is a markdown shortcuts example; it might take a bit of effort (the existing code is only doing block-level shortcuts) but it should be applicable to bold as well.
https://www.slatejs.org/examples/markdown-shortcuts
